src/components/MessageList/TypingIndicator.js

```diff
@@ -24,74 +22,72 @@ const Container = styled.View`
   ${({ theme }) => theme.typingIndicator.container.css};
 `;
 
-class TypingIndicator extends React.PureComponent {
-  static themePath = 'typingIndicator';
-  static propTypes = {
-    typing: PropTypes.object,
-    client: PropTypes.instanceOf(StreamChat),
-    /**
-     * Defaults to and accepts same props as: [Avatar](https://getstream.github.io/stream-chat-react-native/#avatar)
-     */
-    Avatar: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-  };
-  static defaultProps = {
-    Avatar,
-  };
+const constructTypingString = (typingUsers, client, t) => {
+  const typingKeys = Object.keys(typingUsers);
+  const nonSelfUsers = [];
+  typingKeys.forEach((item, i) => {
+    if (client.user.id === typingUsers[typingKeys[i]].user.id) {
+      return;
+    }
+    nonSelfUsers.push(
+      typingUsers[typingKeys[i]].user.name ||
+        typingUsers[typingKeys[i]].user.id,
+    );
+  });
 
-  constructTypingString = (dict) => {
-    const { client, t } = this.props;
-    const arr2 = Object.keys(dict);
-    const arr3 = [];
-    arr2.forEach((item, i) => {
-      if (client.user.id === dict[arr2[i]].user.id) {
-        return;
-      }
-      arr3.push(dict[arr2[i]].user.name || dict[arr2[i]].user.id);
+  let outStr = '';
+  if (nonSelfUsers.length === 1) {
+    outStr = t('{{ user }} is typing...', { user: nonSelfUsers[0] });
+  } else if (nonSelfUsers.length === 2) {
+    //joins all with "and" but =no commas
```

Comment:
`=no`?

Reply:
Haha I'll clean up those comments, I tried to keep everything more or less the same in these to FC moves with the thought that optimization and correction would be in the TS'ing.